corehq/apps/userreports/pillow.py

```diff
@@ -133,3 +149,56 @@ def get_all_configs(self):
     def rebuild_table(self, sql_adapter):
         super(StaticDataSourcePillow, self).rebuild_table(sql_adapter)
         rebuild_indicators.delay(sql_adapter.config.get_id)
+
+
+class ConfigurableReportPillowProcessor(ConfigurableReportTableManagerMixin, PillowProcessor):
+
+    def process_change(self, pillow_instance, change, do_set_checkpoint):
+        self.bootstrap_if_needed()
+        if change.deleted:
+            # we don't currently support hard-deletions at all.
+            # we may want to change this at some later date but seem ok for now.
+            # see https://github.com/dimagi/commcare-hq/pull/6944 for rationale
+            return
+
+        domain = change.metadata.domain
+        if not domain:
+            # if no domain we won't save to any UCR table
+            return
+
+        for table in self.table_adapters:
+            if table.config.domain == domain and KAFKA_UCRS.enabled(domain):
+                # only bother getting the document if we have a domain match from the metadata
+                doc = change.get_document()
+                if table.config.filter(doc):
+                    table.save(doc)
+                elif table.config.deleted_filter(doc):
+                    table.delete(doc)
+
+
+class ConfigurableReportKafkaPillow(ConstructedPillow):
+
+    def __init__(self, pillow_name):
+        change_feed = KafkaChangeFeed(topics.ALL, group_id=pillow_name)
```

Comment:
Sorry, what I meant was that `topics.ALL` includes `topics.META` but I don't think it's needed for this pillow. I imagine the changes will just be ignored but we could also just not listen to that topic.